c7n/resources/asg.py

```diff
@@ -492,6 +492,54 @@ def process(self, asgs, event=None):
         return super(VpcIdFilter, self).process(asgs)
 
 
+@filters.register('launch-failure')
+class LaunchFailureFilter(Filter):
+    """
+    This filter is designed to report ASGs that are constantly trying to
+    spin up new instances & failing to do so over & over.
+    The filter can be used to search for failure events over a period of time
+    by using the 'days' parameter.
+
+    - name: asg-launch-failure
+      resource: asg
+      filters:
+        - launch-failure
+          days: 10
+    """
+    schema = type_schema(
+        'launch-failure',
+        days={'type': 'number'})
+
+    def asg_activities(self, asg):
+        limit_date = datetime.now(
+            tz=tzutc()) - timedelta(days=self.data.get('days', 1))
+        client = local_session(
+            self.manager.session_factory).client('autoscaling')
+        p = client.get_paginator('describe_scaling_activities')
+        results = {}
+        for activity in p.paginate(
+                AutoScalingGroupName=asg['AutoScalingGroupName']):
+            for a in activity['Activities']:
+                if a['StartTime'].date() < limit_date.date():
```

Comment:
it seems a bit wasteful to generate a date object when we already have datetimes, can we just compare the datetime's directly and avoid the temp objects per activity event. you can limit_date.replace(hour=0, minute=0, second=0) if you want to capture the full day's activities.